noir/noir-repo/compiler/noirc_frontend/src/monomorphization/mod.rs

```diff
@@ -246,7 +246,12 @@ impl<'interner> Monomorphizer<'interner> {
                         }
                     }
                     FunctionKind::Recursive => {
-                        unreachable!("Only main can be specified as recursive, which should already be checked");
+                        // let func = self.interner.function_meta(&id);
+                        // println!("{:#?}", func.name);
+                        let id =
+                            self.queue_function(id, expr_id, typ, turbofish_generics, trait_method);
+                        Definition::Function(id)
+                        //unreachable!("wtf");
```

Comment:
This check is preventing us from testing entrypoint functions (and thus, contracts). 

The macro is setting the `.kind` of aztec functions to `Recursive`, (directly and not via the attribute, that's why we didn't see it in the offsite @vezenovm). Any thoughts? CC @TomAFrench 

Can we skip it for test functions somehow?

Reply:
Yeah, I think we can turn off this check for test functions.